progs/incorrect/unbound_var.sched

```diff
@@ -1,5 +1,5 @@
 classes X, Y, Z;
 
-foo = strict[X, Y, Z];
+foo = strict[fifo[X], fifo[Y], fifo[Z]];
```

Comment:
Would `strict[union[X, Y, Z]]` be equivalent?

Reply:
If so, there are some other points in the code that we could talk about. But I won't pollute the PR with too many comments until we've hashed this out.

Reply:
Ah, this could be a gray area that we tidy up at the meeting.

I was under the impression that `strict` is stream-to-stream, but `union` is to be treated as a `set` only – so that it would need to be passed to a set-to-stream transformer before being used as a policy under a stream-to-stream transformer. 

By this, I'm pretty sure that if the parser works correctly, `strict[union[X, Y, Z]]` shouldn't parse.

If that's not how we want it though, I agree that this needs some reworking.